mountpoint-s3/src/inode.rs

```diff
@@ -780,6 +786,8 @@ pub struct InodeStat {
     pub ctime: OffsetDateTime,
     /// Time of last access
     pub atime: OffsetDateTime,
+    /// Etag for the file (object)
+    pub etag: Option<String>,
```

Comment:
We use the `InodeStat` struct here to mean "cached copy of remote metadata", which is maybe a bad name.

Reply:
I see, maybe that makes sense... I'm a little confused by this module comment.

```rs
//! In addition to this "permanent" state, an [Inode] also has some cached state called [InodeStat].
//! Cached state is subject to an expiry time, and must be refreshed before use if it has expired.
//! Some cached state is dependent on the inode kind; that state is hidden behind a [InodeStatKind]
//! enum.
```

Still sounds like the etag belongs in the `InodeKindData::File` enum, no?

Reply:
Yes, probably. I am already setting it to None in case of Directory. Should I add it only in the `InodeKindData::File` like children in `InodeKindData::Directory`? And in that case, should I make `InodeKindData` public as I am not sure I will be able to use etag from it elsewhere without making it pub?

Reply:
That comment is out of date, `﻿﻿InodeStatKind` doesn't exist any more. We used to separate stat metadata by inode type, but we don't any more. Perhaps we should again, but for now the etag should just live in `InodeStat`. It's better that we keep "all state that might be remotely changed" in one place, which is what `InodeStat` is right now.

Reply:
OK that makes sense.

Yeah, maybe name change makes sense or at least some doc to accompany it. It can be in another PR.

Reply:
@jamesbornholt What kind of attributes should be there in `InodeKindData` in that case?

Reply:
`InodeKindData` is for local state. Right now that's just "list of children" for directories, and nothing for files.